### PR TITLE
Origin/add discovery

### PR DESCRIPTION
**Pull request by kqpham**
### What this PR does (required):
- Added in the contest integration to discover so it shows to both logged in users and non users
- Added in conditional functionality that will render specific buttons for logged in user
- Added change in pathing for contest within dashboard
- updated contest routes

### Screenshots / Videos (required):
- Post at least one screenshot to show the feature in action

https://user-images.githubusercontent.com/17209997/122124636-8e22d700-cdfd-11eb-8113-249200842bad.mov

### Any information needed to test this feature (required):
- The steps taken to test this feature is to log into an existing user and go to the home page to view all current contests.
- clicking one of the contest to see if it goes to the specific contest url
- Attempting to go to the home page without logging in to see al the contests
- clicking on one of the contest to see if the end-user is redirected to a sign in page.

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

File: client/src/pages/Dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -30,22 +30,34 @@ export default function Dashboard(): JSX.Element {
...
+      sm={12}
+      md={12}
+      direction="column"
+      justify="flex-start"
+      component="main"
+      className={`${classes.root}`}
+    >
       <CssBaseline />
-      <Grid item className={classes.drawerWrapper}>
+      <Grid className={classes.drawerWrapper}>
```
Comment (dproc96) [inline client/src/pages/Dashboard/Dashboard.tsx:45]: Why did you remove the `item` here?

File: server/controllers/submission.js [hunk 2]
```diff
@@ -74,7 +74,6 @@ exports.createSubmission = asyncHandler(async (req, res, next) => {
 });
 
 exports.getAllSubmissions = asyncHandler(async (req, res, next) => {
-    console.log(req.params);
     const contest_id = req.params.id;
     const Creator = req.params.creator;
     console.log("contest id",contest_id);
```
Comment (dproc96) [inline server/controllers/submission.js:79]: You can remove this console log too

**Review by dproc96 — APPROVED** [review]
This looks good, just one question and one small note

File: client/src/pages/Dashboard/Dashboard.tsx (hunk 1 above)
Comment (kqpham) [reply client/src/pages/Dashboard/Dashboard.tsx:45]: I remove the item there due to some styling issues that were conflicting within item's declared styling.

File: server/controllers/submission.js (hunk 2 above)
Comment (kqpham) [reply server/controllers/submission.js:79]: Yeah I will remove that right away, that one slipped away from me!
